app/services/rapid_ready_for_decision/hypertension_upload_manager.rb

```diff
@@ -4,15 +4,25 @@ module RapidReadyForDecision
   class HypertensionUploadManager
     attr_accessor :submission
 
+    HYPERTENSION_PDF_NAME = 'VAMC_Hypertension_Rapid_Decision_Evidence'
+
     def initialize(submission)
       @submission = submission
+      @date = Time.zone.today
+    end
+
+    def file_upload_name
+      end_date = @date.strftime('%m/%d/%Y')
+      start_date = (@date - 1.year).strftime('%m/%d/%Y')
+
+      "#{HYPERTENSION_PDF_NAME}[#{start_date}-#{end_date}].pdf"
```

Comment:
`@date` is only used within one method, so:
```suggestion
    end

    def file_upload_name
      today = Time.zone.today
      end_date = today.strftime('%m/%d/%Y')
      start_date = (today - 1.year).strftime('%m/%d/%Y')

      "#{HYPERTENSION_PDF_NAME}[#{start_date}-#{end_date}].pdf"
```
Why is `start_date` 1 year ago (FYI: you could use `1.year.ago` instead)? Should it be the submission date?

Reply:
@yoomlam So... yes, you're right. _But_ I had some weird reasoning based on early naive thinking about jobs and asynchrony that might be interesting: the actual date/time when processing starts is when the class is initialized. I assumed this was the "most accurate" because that date/time could be referenced later by a job. But upon further examination of the actual flow, this doesn't make any sense...

Anyway, I wrote this out mostly for myself but now I have a better understanding of how things fit together. (Seeing some relevant comments downstream, too).

Reply:
Hmmm. I was considering suggesting the `submission.updated_at` date but there is a chance the job may retry over several days. Ideally, we should use the time that the lighthouse query is performed. We could somehow either record that timestamp or send it to this class, but I don't know if that additional complexity is worth the increased precision. I'd revisit the ticket motivating this PR to see how precise we need to be.

Reply:
> Hmmm. I was considering suggesting the `submission.updated_at` date but there is a chance the job may retry over several days. Ideally, we should use the time that the lighthouse query is performed. We could somehow either record that timestamp or send it to this class, but I don't know if that additional complexity is worth the increased precision. I'd revisit the ticket motivating this PR to see how precise we need to be.

Yes, I was thinking exactly the same thing. I did some digging yesterday and the spirit of this change is to add some level of uniqueness to the PDF filename. I'm pinging researchers/designers on it to make sure! Still, we'd ideally grab the time the query was performed.